examples/build/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	v1 "github.com/openshift/api/build/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	buildv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
+	"path/filepath"
+	"flag"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"fmt"
+)
+
+func main() {
+	var kubeconfig *string
+	if home := homeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	buildV1Client, err := buildv1.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	namespace := "testproject"
+	// get all builds
+	builds, err := buildV1Client.Builds(namespace).List(metav1.ListOptions{})
+
+	if err != nil {
+		panic(err.Error())
+	}
+	fmt.Printf("There are %d builds in project %s\n", len(builds.Items), namespace)
+
+	// List names of all builds
+	for i, build := range builds.Items {
+		fmt.Printf("index %d: Name of the build: %s\n", i, build.Name)
+	}
+
+	// get a specific build
+	build := "cakephp-ex-1"
+	myBuild, err := buildV1Client.Builds(namespace).Get(build, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
```

Comment:
switch/case to match project style.

Reply:
After adding start() function with error returned, I thought it could be simplified by just treating all types of error by panic.